### PR TITLE
Add cmake macros to simplify cmake files

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

File: tesseract/tesseract_common/cmake/tesseract_macros.cmake [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#
+# @file tesseract_macros.cmae
+# @brief Common Tesseract CMake Macros
+#
+# @author Levi Armstrong
+# @date January 18, 2018
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_common/cmake/tesseract_macros.cmake]: Updated

File: tesseract/tesseract_common/cmake/tesseract_macros.cmake [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# @par
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+macro(tesseract_target_compile_options target)
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_common/cmake/tesseract_macros.cmake]: Documentation Added.

**Comment by Levi-Armstrong** [comment]
@mpowelson changes look good?

**Comment by mpowelson** [comment]
Yeah it looks good to me

**Pull request merged**